src/NUnitFramework/framework/Constraints/NUnitEqualityComparer.cs

```diff
@@ -193,7 +205,7 @@ public bool AreEqual(object? x, object? y, ref Tolerance tolerance)
             static string GetType(object? x) => x?.GetType().FullName ?? "null";
```

Comment:
Thanks for catching this. We may need to keep the `GetType()` rather than switch to `typeof()` as this is called from more than just the equal constraint, but a generic version would certainly be a welcome addition.